client/core/core.go

```diff
@@ -7046,3 +7047,140 @@ func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFi
+	}
+
+	return txID, nil
+}
+
+// AccelerationEstimate returns the amount of funds that would be needed to
+// accelerate the swap transactions in an order to a desired fee rate.
+func (c *Core) AccelerationEstimate(oidB dex.Bytes, newFeeRate uint64) (uint64, error) {
+	tracker, swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, err := c.orderAccelerationParameters(oidB)
+	if err != nil {
+		return 0, err
+	}
+
+	accelerationFee, err := tracker.wallets.fromWallet.accelerationEstimate(swapCoins, accelerationCoins, changeCoin, requiredForRemainingSwaps, newFeeRate)
+	if err != nil {
+		return 0, err
+	}
+
+	return accelerationFee, nil
+}
+
+// PreAccelerateOrder returns information the user can use to decide how much
+// to accelerate stuck swap transactions in an order.
+func (c *Core) PreAccelerateOrder(oidB dex.Bytes) (*PreAccelerate, error) {
+	tracker, swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, err := c.orderAccelerationParameters(oidB)
+	if err != nil {
+		return nil, err
+	}
+
+	feeSuggestion := c.feeSuggestionAny(tracker.fromAssetID)
+
+	currentRate, suggestedRange, err :=
+		tracker.wallets.fromWallet.preAccelerate(swapCoinIDs, accelerationCoins, changeCoinID, requiredForRemainingSwaps, feeSuggestion)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PreAccelerate{
+		SwapRate:       currentRate,
+		SuggestedRate:  feeSuggestion,
+		SuggestedRange: suggestedRange,
+	}, nil
+}
+
+// orderAccelerationParameters takes an order id, and returns the parameters
+// needed to accelerate the swap transactions in that order.
+func (c *Core) orderAccelerationParameters(oidB dex.Bytes) (tracker *trackedTrade, swapCoins, accelerationCoins []dex.Bytes, changeCoin dex.Bytes, requiredForRemainingSwaps uint64, err error) {
+	makeError := func(err error) (*trackedTrade, []dex.Bytes, []dex.Bytes, dex.Bytes, uint64, error) {
+		return nil, nil, nil, nil, 0, err
+	}
+
+	if len(oidB) != order.OrderIDSize {
+		return makeError(fmt.Errorf("wrong order ID length. wanted %d, got %d", order.OrderIDSize, len(oidB)))
+	}
+	var oid order.OrderID
+	copy(oid[:], oidB)
+
+	for _, dc := range c.dexConnections() {
+		tracker, _, _ = dc.findOrder(oid)
+		if tracker != nil {
+			mktID := marketName(tracker.Base(), tracker.Quote())
+			mkt := dc.marketConfig(mktID)
+			if mkt == nil {
+				return makeError(fmt.Errorf("could not find market: %v", mktID))
+			}
+			lotSize := mkt.LotSize
+			fromAsset := dc.assets[tracker.fromAssetID]
+			if fromAsset == nil {
+				return makeError(fmt.Errorf("could not find asset with id: %v", tracker.fromAssetID))
+			}
+			swapSize := fromAsset.SwapSize
+			lotsRemaining := tracker.Trade().Remaining() / lotSize
+			requiredForRemainingSwaps = lotsRemaining * swapSize * tracker.metaData.MaxFeeRate
+			break
+		}
+	}
+	if tracker == nil {
+		return makeError(fmt.Errorf("could not find active order with id: %s", oid))
+	}
+
+	tracker.mtx.Lock()
+	defer tracker.mtx.Unlock()
```

Comment:
Can be RLock() because only reading here?

But actually, I think this lock needs to be moved up a scope, so nothing will touch the order while we swap the change coin.